src/Constraint.jl

```diff
@@ -3,17 +3,17 @@
 # Use of this source code is governed by a BSD-style license that can be found
 # in the LICENSE file or at https://opensource.org/license/bsd-2-clause
 
-mutable struct Constraint{S<:MOI.AbstractSet}
+mutable struct Constraint{S<:MOI.AbstractVectorSet}
```

Comment:
Why is this needed?

Reply:
It's not needed, I just thought we are only supporting vector sets at the moment, we can revert that part

Reply:
For example, isn't `populate_dual!` always returning a vector ?

Reply:
Ah true. We can always extend this non-breaking in future.

Reply:
I guess there'd be an error like this at the moment:
```julia
julia> reshape(2.0, (1,1))
ERROR: MethodError: no method matching reshape(::Float64, ::Tuple{Int64, Int64})
```

So we'd need a scalar `populate_dual!` method, but that is do-able (once there is demand).